File: 832. Flipping an Image.go
// https://leetcode.com/problems/flipping-an-image/

package leetcode

func flipAndInvertImage(A [][]int) [][]int {
	result := make([][]int, len(A))

	for r, row := range A {
		result[r] = make([]int, len(row))
		rowLen := len(row)
		for c := rowLen - 1; c >= 0; c-- {
			offset := rowLen - c - 1
			result[r][offset] = (^row[c] & 1)
		}
	}

	return result
}

File: 832. Flipping an Image.js
// https://leetcode.com/problems/flipping-an-image/

/**
 * @param {number[][]} A
 * @return {number[][]}
 */
var flipAndInvertImage = function(A) {
    return new Array(A.length).fill(0)
    .map((_, r) => A[r].map((_, c) => ~A[r][A[0].length - c - 1] & 1));
};